packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -138,20 +167,33 @@ class _BottomAppBarState extends State<BottomAppBar> {
           notchMargin: widget.notchMargin,
         )
       : const ShapeBorderClipper(shape: RoundedRectangleBorder());
-    final double elevation = widget.elevation ?? babTheme.elevation ?? _defaultElevation;
-    final Color color = widget.color ?? babTheme.color ?? Theme.of(context).bottomAppBarColor;
-    final Color effectiveColor = ElevationOverlay.applyOverlay(context, color, elevation);
-    return PhysicalShape(
-      clipper: clipper,
-      elevation: elevation,
-      color: effectiveColor,
-      clipBehavior: widget.clipBehavior,
-      child: Material(
-        key: materialKey,
-        type: MaterialType.transparency,
-        child: widget.child == null
-          ? null
-          : SafeArea(child: widget.child!),
+    final double elevation = widget.elevation ?? babTheme.elevation ?? defaults.elevation!;
+    final double? height = widget.height ?? babTheme.height ?? defaults.height;
+    final Color color = widget.color ?? babTheme.color ?? defaults.color!;
+    final Color surfaceTintColor = widget.surfaceTintColor ?? babTheme.surfaceTintColor ?? defaults.surfaceTintColor!;
+    final Color effectiveColor = isMaterial3 ? color : ElevationOverlay.applyOverlay(context, color, elevation);
+
+    final Widget? child = isMaterial3 ? Padding(
+      padding: const EdgeInsets.symmetric(vertical: 12.0, horizontal: 16.0),
+      child: widget.child,
+    ) : widget.child;
+
+    return SizedBox(
+      height: height,
+      child: PhysicalShape(
+        clipper: clipper,
+        elevation: elevation,
+        color: effectiveColor,
+        clipBehavior: widget.clipBehavior,
+        child: Material(
+          key: materialKey,
+          type: isMaterial3 ? MaterialType.canvas : MaterialType.transparency,
+          elevation: elevation,
```

Comment:
I notice that we are adding the `elevation` to both the `PhysicalShape` and the `Material` widget. Will this cause any problem with it being handled by both? Or is this just to get the surface tint to work properly on the material widget?